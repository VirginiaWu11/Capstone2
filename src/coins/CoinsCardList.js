import React, { memo } from "react";
import Grid from "@mui/material/Grid";
import CoinCard from "./CoinCard";
import { CardActionArea } from "@mui/material";
import Button from "@mui/material/Button";

const CoinsCardList = memo(
  ({
    coins,
    handleCoinModalOpen,
    isPinned,
    handlePin,
    handleUnpin,
    isOnPortfolio,
    removeFromPortfolio,
    handlePortfolioModalOpen,
  }) => {
    console.debug("CoinsCardList rendered");
    return (
      <div>
        <Grid
          container
          spacing={{ xs: 2, md: 3 }}
          columns={{ xs: 3, sm: 6, md: 9, lg: 12 }}
          justifyContent="center"
          p={2}
        >
          {coins.map((coin) => (
            <Grid key={coin.id} item xs={3} sm={3} md={3} lg={3}>
              <CardActionArea
                component={Button}
                onClick={() => {
                  handleCoinModalOpen(coin);
                }}
              >
                <CoinCard
                  key={coin.id}
                  coin={coin}
                  isPinned={isPinned}
                  handlePin={handlePin}
                  handleUnpin={handleUnpin}
                  isOnPortfolio={isOnPortfolio}
                  removeFromPortfolio={removeFromPortfolio}
                  handlePortfolioModalOpen={handlePortfolioModalOpen}
                />{" "}
              </CardActionArea>
            </Grid>
          ))}{" "}
        </Grid>
      </div>
    );
  }
);

export default CoinsCardList;
