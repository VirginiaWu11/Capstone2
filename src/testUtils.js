import React from "react";
import { UserContext } from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
};

const UserProvider = ({
  children,
  currentUser = demoUser,
  setCurrentUser,
  token,
  setToken,
  signup,
  signin,
  signout,
  updateProfile,
  infoLoaded,
  setInfoLoaded,
}) => (
  <UserContext.Provider
    value={{
      currentUser,
      setCurrentUser,
      token,
      setToken,
      signup,
      signin,
      signout,
      updateProfile,
      infoLoaded,
      setInfoLoaded,
    }}
  >
    {children}
  </UserContext.Provider>
);

export { UserProvider };
