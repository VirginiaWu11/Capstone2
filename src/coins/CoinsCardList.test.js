import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import CoinsCardList from "./CoinsCardList";
import { UserProvider } from "../testUtils";

it("matches snapshot", function () {
  let coins = [];
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <CoinsCardList coins={coins} />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
