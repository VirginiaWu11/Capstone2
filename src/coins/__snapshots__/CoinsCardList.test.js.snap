// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-spacing-md-3 css-wto7gs-MuiGrid-root"
    >
       
    </div>
  </div>
</DocumentFragment>
`;
